### PR TITLE
Fwxv 255 moved I2C to PCA

### PR DESCRIPTION
**Pull request by warren-xu**

### CODE REVIEW

**Comment by warren-xu** [comment]
accidentally included gpio_leds folder in the pull request, can ignore

File: libraries/ms-drivers/src/pca9555_gpio_expander.c [hunk 1]
```diff
@@ -135,3 +135,23 @@ StatusCode pca9555_gpio_subscribe_interrupts(const GpioAddress *interrupt_pin, E
   gpio_it_register_interrupt(interrupt_pin, &interrupt_settings, event, task);
   return STATUS_CODE_OK;
 }
+
+StatusCode pca9555_get_register_state(I2CAddress addr, uint16_t *reg_val) {
+  if (addr >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid PCA9555 port.");
+  }
+  uint8_t reg_arr[2] = {};       // store read value
+  uint8_t reg_to_read = INPUT0;  // read the first register
```
Comment (JarvisWeng) [inline libraries/ms-drivers/src/pca9555_gpio_expander.c:144]: What is `INPUT0` and `INPUT1`? Can't seem to find them.
But basically you'll always only read and write 1 byte?

File: projects/gpio_leds/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!--
```
Comment (JarvisWeng) [inline projects/gpio_leds/README.md:1]: You should probably get rid of this folder

File: libraries/ms-drivers/src/pca9555_gpio_expander.c (hunk 1 above)
Comment (warren-xu) [reply libraries/ms-drivers/src/pca9555_gpio_expander.c:144]: I was trying to find at what address the PCA9555 registers are at, but I couldn't find them so I assumed it was INPUT0 and INPUT1. Do you know where I should be reading from? 
Yeah, essentially the function will read and write one byte for both registers and output them as a 16-bit int.

Comment (Bafran) [reply libraries/ms-drivers/src/pca9555_gpio_expander.c:144]: If you look at the datasheet pg 19, it should be there

File: libraries/ms-drivers/src/pca9555_gpio_expander.c [hunk 3]
```diff
@@ -135,3 +135,23 @@ StatusCode pca9555_gpio_subscribe_interrupts(const GpioAddress *interrupt_pin, E
   gpio_it_register_interrupt(interrupt_pin, &interrupt_settings, event, task);
   return STATUS_CODE_OK;
 }
+
+StatusCode pca9555_get_register_state(I2CAddress addr, uint16_t *reg_val) {
+  if (addr >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid PCA9555 port.");
+  }
+  uint8_t reg_arr[2] = {};     // store read value
+  uint8_t reg_to_read = 0x00;  // read the first register
```
Comment (JarvisWeng) [inline libraries/ms-drivers/src/pca9555_gpio_expander.c:144]: The regs to read are 0x0 and 0x1? Could you make them into #defines so they're not magic numbers?

**Comment by Bafran** [comment]
Closing out for now, please let us know if it should get merged

**Pull request closed without merging**